hoomd/hpmc/ComputeSDF.h

```diff
@@ -264,27 +269,64 @@ template<class Shape> void ComputeSDF<Shape>::zeroHistogram()
     // Zero the histogram
     for (size_t i = 0; i < m_hist.size(); i++)
         {
-        m_hist[i] = 0;
+        m_hist[i] = 0.0;
         }
     }
 
+template<class Shape> Scalar ComputeSDF<Shape>::getMaxInteractionDiameter()
+    {
+    Scalar max_core_diameter = m_mc->getMaxCoreDiameter();
+    Scalar max_r_cut_patch(0.0);
+
+    if (m_mc->getPatchEnergy())
+        {
+        const std::vector<param_type, hoomd::detail::managed_allocator<param_type>>& params
+            = m_mc->getParams();
+        for (unsigned int typ_i = 0; typ_i < m_pdata->getNTypes(); typ_i++)
+            {
+            Shape shape_i(quat<Scalar>(), params[typ_i]);
+            Scalar r_cut_patch_i = m_mc->getPatchEnergy()->getRCut()
+                                   + 0.5 * m_mc->getPatchEnergy()->getAdditiveCutoff(typ_i);
+            max_r_cut_patch = std::max(max_r_cut_patch, r_cut_patch_i);
+            }
+        }
+    return std::max(max_core_diameter, max_r_cut_patch);
+    }
+
 /*! \param timestep current timestep
 
     countHistogram() loops through all particle pairs *i,j* where *i* is on the local rank, computes
-    the bin in which that pair should be and adds 1 to the bin. countHistogram() can be called
-    multiple times to increment the counters for averaging, and it operates without any
-    communication.
+    the bin corresponding to the scale factor of the vector r_ij that induces the first overlap
+    between particles *i* and *j*, and then adds s_ij to the bin, where s_ij = 1 - e^{\beta\Delta
+    U_{ij}}. Note that this definition allows for the consideration of soft overlaps where the pair
+    potential changes value, and reduces to 1 for hard particle overlaps.  countHistogram() operates
+    without any communication.
       - The integrator performs the ghost exchange (with the ghost width extra that we add)
-      - Only on writeOutput() do we need to sum the per-rank histograms into a global histogram
+
+    This function is a wrapper that calls the appropriate method depending on the value of
+    m_mc->getPatchEnergy()
 */
 template<class Shape> void ComputeSDF<Shape>::countHistogram(uint64_t timestep)
+    {
+    if (m_mc->getPatchEnergy())
+        {
+        countHistogramLinearSearch(timestep);
+        }
+    else
+        {
+        countHistogramBinarySearch(timestep);
+        }
+    } // end countHistogram()
+
+template<class Shape> void ComputeSDF<Shape>::countHistogramBinarySearch(uint64_t timestep)
     {
     // update the aabb tree
     const hoomd::detail::AABBTree& aabb_tree = m_mc->buildAABBTree();
+
     // update the image list
     const std::vector<vec3<Scalar>>& image_list = m_mc->updateImageList();
 
-    Scalar extra_width = m_xmax / (1 - m_xmax) * m_mc->getMaxCoreDiameter();
+    Scalar extra_width = m_xmax / (1 - m_xmax) * this->getMaxInteractionDiameter();
```

Comment:
We could save a few cycles by making `max_diam` a member variable (`m_max_diam`) and reusing the computation here.